### PR TITLE
[CI] Send message for near adherents with sames interests

### PR DESCRIPTION
**Pull request by PaskR**

### CODE REVIEW

File: src/Mailjet/Message/CitizenInitiativeAdherentsNearMessage.php [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+use AppBundle\Entity\CitizenInitiative;
+use Ramsey\Uuid\Uuid;
+
+class CitizenInitiativeAdherentsNearMessage extends MailjetMessage
+{
+    /**
+     * Creates a new message instance for a list of recipients.
+     *
+     * @param Adherent[]        $recipients
+     * @param Adherent          $host
```
Comment (illusionOfParadise) [inline src/Mailjet/Message/CitizenInitiativeAdherentsNearMessage.php]: It's better to change the name of variable, it's not a host but an organizer

Comment (PaskR) [inline src/Mailjet/Message/CitizenInitiativeAdherentsNearMessage.php]: Okay, I'll do it

File: app/config/services/platform.xml [hunk 2]
```diff
@@ -234,6 +234,13 @@
             <argument type="service" id="app.mailjet.transactional_mailer"/>
         </service>
 
+        <service id="app.citizen_initiative.message_notifier" class="AppBundle\CitizenInitiative\CitizenInitiativeMessageNotifier">
```
Comment (HeahDude) [inline app/config/services/platform.xml]: We now use the class as id.

File: src/CitizenInitiative/CitizenInitiativeMessageNotifier.php [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        $this->adherentManager = $adherentManager;
+        $this->urlGenerator = $urlGenerator;
+    }
+
+    public function onCitizenInitiativeValidated(EventValidatedCitizenInitiative $event): void
+    {
+        $this->sendMessageToAdherentsNearWithSameInterests($event);
+    }
+
+    private function sendMessageToAdherentsNearWithSameInterests(EventValidatedCitizenInitiative $event): void
```
Comment (HeahDude) [inline src/CitizenInitiative/CitizenInitiativeMessageNotifier.php]: `sendMessageToNearByAdherentsWithSameInterests`

File: src/CitizenInitiative/CitizenInitiativeMessageNotifier.php [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            $this->adherentManager->findNearByCitizenInitiativeInterests($event->getCitizenInitiative())->toArray(),
+            MailjetService::PAYLOAD_MAXSIZE
+        );
+
+        foreach ($chunks as $chunk) {
+            $this->mailjet->sendMessage($this->createMessageToAdherentsNearWithSameInterests($chunk, $event->getCitizenInitiative(), $event->getAuthor()));
+        }
+    }
+
+    private function createMessageToAdherentsNearWithSameInterests(array $adherents, CitizenInitiative $citizenInitiative, Adherent $organizer): CitizenInitiativeAdherentsNearMessage
```
Comment (HeahDude) [inline src/CitizenInitiative/CitizenInitiativeMessageNotifier.php]: `createMessageToNearByAdherentsWithSameInterests`

File: src/Event/EventValidatedCitizenInitiative.php [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    private $author;
+    private $initiative;
+
+    public function __construct(Adherent $author, CitizenInitiative $initiative)
+    {
+        $this->author = $author;
+        $this->initiative = $initiative;
+    }
+
+    public function getAuthor()
```
Comment (HeahDude) [inline src/Event/EventValidatedCitizenInitiative.php]: Missing return type hint.

File: src/Mailjet/Message/CitizenInitiativeAdherentsNearMessage.php [hunk 6]
```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace AppBundle\Mailjet\Message;
+
+use AppBundle\Entity\Adherent;
+use AppBundle\Entity\CitizenInitiative;
+use Ramsey\Uuid\Uuid;
+
+class CitizenInitiativeAdherentsNearMessage extends MailjetMessage
```
Comment (HeahDude) [inline src/Mailjet/Message/CitizenInitiativeAdherentsNearMessage.php]: Should be final.

File: src/Mailjet/Message/CitizenInitiativeAdherentsNearMessage.php [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+        if (!$recipients) {
+            throw new \InvalidArgumentException('At least one Adherent recipients is required.');
+        }
+
+        $recipient = array_shift($recipients);
+        if (!$recipient instanceof Adherent) {
+            throw new \RuntimeException('First recipient must be an Adherent instance.');
+        }
+
+        $vars = static::getTemplateVars(
```
Comment (HeahDude) [inline src/Mailjet/Message/CitizenInitiativeAdherentsNearMessage.php]: `self` should be used everywhere in this final class.

File: src/Query/Mysql/JsonContains.php [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+     * @return string
+     *
+     * @throws DBALException
+     */
+    public function getSql(SqlWalker $sqlWalker): string
+    {
+        $jsonDoc = $sqlWalker->walkStringPrimary($this->jsonDocExpr);
+        $jsonVal = $sqlWalker->walkStringPrimary($this->jsonValExpr);
+        $jsonPath = '';
+        if (!empty($this->jsonPathExpr)) {
```
Comment (HeahDude) [inline src/Query/Mysql/JsonContains.php]: `if ($this->jsonPathExpr)`

File: src/Repository/AdherentRepository.php [hunk 9]
```diff
@@ -239,4 +245,25 @@ public function findByEvent(Event $event): AdherentCollection
...
+
+    public function findNearByCitizenInitiativeInterests(CitizenInitiative $citizenInitiative): AdherentCollection
+    {
+        $qb = $this
+            ->createNearbyQueryBuilder(new Coordinates($citizenInitiative->getLatitude(), $citizenInitiative->getLongitude()))
+            ->andWhere($this->getNearbyExpression().' <= :distance_max')
+            ->setParameter('distance_max', self::CITIZEN_INITIATIVE_RADIUS);
+
+        if (false === empty($interests = $citizenInitiative->getInterests())) {
+            $conditions = [];
```
Comment (HeahDude) [inline src/Repository/AdherentRepository.php]: You can remove this line and use `$orX->addMultiple($conditions ?? []);` below.

File: src/CitizenInitiative/AbstractCitizenInitiativeEvent.php [hunk 10]
```diff
@@ -1,12 +1,12 @@
 <?php
 
-namespace AppBundle\Event;
+namespace AppBundle\CitizenInitiative;
 
+use AppBundle\Committee\CommitteeEvent as BaseEvent;
```
Comment (illusionOfParadise) [inline src/CitizenInitiative/AbstractCitizenInitiativeEvent.php]: No need to extend this class

**Pull request merged**